extensions/vertx/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java

```diff
@@ -381,7 +381,10 @@ private static int calculateDefaultIOThreads() {
         //it's hard to say what this number should be, but it is also obvious
         //that for constrained environments we don't want a lot of event loops
         //lets start with 10mb and adjust as needed
-        int recommended = ProcessorInfo.availableProcessors() * 2;
+        //We used to recommend a default of twice the number of cores,
+        //but more recent developments seem to suggest matching the number of cores 1:1
+        //being a more reasonable default. It also saves memory.
```

Comment:
The `*2` factor was because of hyperthreading on bare metal. Yes, it does not apply to containers, and with the high number of cores we have today, this factor does not matter anymore. 

However, I would suggest adding that to the comment:

- if you are using the application on bare metal and want to leverage hyperthreading, you should use the number of CPU cores * 2
- in containers or if the number of CPU cores is large enough for the need of your application, having the number of event loops matching the number of CPU cores is generally better.

Reply:
I think this should be added to the configuration property configuring the number of event loops.

Reply:
> The *2 factor was because of hyperthreading on bare metal. Yes, it does not apply to containers, and with the high number of cores we have today, this factor does not matter anymore.

There's a patch backported to JDK 8 too that now correctly count the number of cores as by number of "logical one" and not physical already ie 2 HT cores are correctly reported by `availableProcessors` as 2
And related containers, now JDK 8 versions too correctly interact with cgroup (and not only, with taskset/numactl as well) to detect the right number of cores.
In the past this wasn't true, that's why wildfly was bothering itself reading *nix paths to verify it

Reply:
welcome back @cescoffier :)

Like @franz1981 said, if the "2x" approach was based on HT then that really doesn't apply anymore since a long time. I guess that explains some of the puzzles we faced.

BTW Franz's reference to WildFly relates to the fact that it has a library directly reading the /proc filesytem to mask additional cores based on cgroups settings; that's also no longer applicable. We're still invoking that library here - but wildfly-commons has been patched to use multi-release jars so it's not doing this anymore when detecting JDK11+ (which is our baseline) - so this is also already not applicable, and I plan to clean it up soon.

Regarding adding such notes to the reference of the configuration setting, are you sure you want that? I certainly could but it seems overly verbose to me: as a user I'd want to know _what_ is the default, and have an option to override, but it's not a tuning guide. 

Perhaps we should work on a tuning guide? @tqvarnst also suggested this; we could expand more on the reasoning there, that would certainly be interesting and we'd have a bit more space to expand on it rather than dropping a mysterious teaser in a single sentence :)

Reply:
Vert.x also did the /proc trick, and yes, in that case, we needed to do the 2x thing.
However, on my R-PI (when I had one), the perf were much better using the 2x factor than the number of cores (bare metal, no container). 

We should still mention to the user how to set up the number of event loops to reduce or increase it in the Vertx Config class.

Reply:
> Vert.x also did the /proc trick, and yes, in that case, we needed to do the 2x thing. However, on my R-PI (when I had one), the perf were much better using the 2x factor than the number of cores (bare metal, no container).

Sure I believe you all when you say it used to be better - but it's no longer the case, and we care for memory too.

> We should still mention to the user how to set up the number of event loops to reduce or increase it in the Vertx Config class.

It's there already?

Reply:
Yes, in `io.quarkus.vertx.core.runtime.config.VertxConfiguration`:

```
    /**
     * The number of event loops. 2 x the number of core by default.
     */
    @ConfigItem
    public OptionalInt eventLoopsPoolSize;
```

That's where we need to mention the change and maybe a bit of discussion.

Reply:
That's already patched in this PR - I'd not have a "discussion" there as I mentioned above as this is a reference.

Could we have explanations and discussion in a different guide? As mentioned this is just a default - and the know isn't a full tuning guide.

Reply:
I agree.